### PR TITLE
(11) feat: set correctly the NetworkSettingsFlowStack

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- Conditional initialization of network settings flow stack

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/screens/NetworkSettings/NetworkSettingsDisclaimerScreen.js [hunk 1]
```diff
@@ -30,6 +31,10 @@ const style = StyleSheet.create({
 export const NetworkSettingsDislaimerNav = Symbol('NetworkSettingsDislaimer').toString();
 
 export function NetworkSettingsDislaimerScreen({ navigation }) {
+  const handleNavigation = () => {
+    navigation.push(NetworkPreSettingsNav);
+  };
```
Comment (andreabadesso) [inline src/screens/NetworkSettings/NetworkSettingsDisclaimerScreen.js:36]: Can we do this?
```suggestion
  const handleNavigation = () => navigation.push(NetworkPreSettingsNav);
```

**Review by andreabadesso — APPROVED** [review]

File: src/screens/NetworkSettings/NetworkSettingsFlowStack.js [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
+  const networkSettings = useSelector((state) => state.networkSettings);
+  const initialNav = isCustomNetwork(networkSettings)
+    ? NetworkPreSettingsNav
+    : NetworkSettingsDislaimerNav;
+
   const FlowStack = createStackNavigator();
   return (
     <FlowStack.Navigator
-      initialRouteName={CustomNetworkSettingsNav}
+      initialRouteName={initialNav}
```
Comment (pedroferreira1) [inline src/screens/NetworkSettings/NetworkSettingsFlowStack.js:24]: We also need a warning and confirmation when changing to the testnet. Do we have it?

Comment (alexruzenhack) [reply src/screens/NetworkSettings/NetworkSettingsFlowStack.js:24]: I didn't understood. Can you clarify? Let me tell what I understood:
- You are saying we need to show the "disclaimer" screen even if the user is using the "testnet" network?
- You are saying we need to take a confirmation when the user pick the "testnet" pre-settings?

File: src/screens/NetworkSettings/NetworkSettingsDisclaimerScreen.js (hunk 1 above)
Comment (alexruzenhack) [reply src/screens/NetworkSettings/NetworkSettingsDisclaimerScreen.js]: - Change lambda to inline expression: c9bee3624619955be9868f29a21e98312a37cd7a

File: src/screens/NetworkSettings/NetworkSettingsFlowStack.js (hunk 2 above)
Comment (alexruzenhack) [reply src/screens/NetworkSettings/NetworkSettingsFlowStack.js:24]: Oh! Ok. I got it now. You are saying we need to show the disclaimer screen for all the users, not matter the networkSettings state. Sure, this is a simpler solution.

Comment (alexruzenhack) [reply src/screens/NetworkSettings/NetworkSettingsFlowStack.js]: - Make disclaimer screen required for any network state: 225c9b53d904051e7a2e23505e4b1364a09b0ce5

File: src/screens/NetworkSettings/NetworkSettingsFlowStack.js [hunk 3]
```diff
@@ -5,11 +5,11 @@ import { NetworkSettingsDislaimerNav, NetworkSettingsDislaimerScreen } from './N
 
 export const NetworkSettingsFlowNav = Symbol('NetworkSettingsFlowStack').toString();
 
-export const NetworkSettingsFlowStack = ({ navigation }) => {
+export const NetworkSettingsFlowStack = () => {
   const FlowStack = createStackNavigator();
   return (
     <FlowStack.Navigator
-      initialRouteName={CustomNetworkSettingsNav}
+      initialRouteName={NetworkSettingsDislaimerNav}
```
Comment (pedroferreira1) [inline src/screens/NetworkSettings/NetworkSettingsFlowStack.js:12]:
```suggestion
      initialRouteName={NetworkSettingsDisclaimerNav}
```

Comment (pedroferreira1) [inline src/screens/NetworkSettings/NetworkSettingsFlowStack.js:12]: This is an error you have already fixed in other PRs, is it missing a rebase?

Comment (alexruzenhack) [reply src/screens/NetworkSettings/NetworkSettingsFlowStack.js:12]: Answered above.

**Comment by alexruzenhack** [comment]
Rebase to resolve conflicts.

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**